packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts

```diff
@@ -33,10 +33,449 @@ export interface IVirtualNode extends cdk.IResource {
+
+  /**
+   * Health checking strategy upstream nodes should use when communicating with the listener
+   *
+   * @default - no healthcheck
+   */
+  readonly healthCheck?: HealthCheck;
+
+  /**
+   * Listener timeout property for HTTP protocol
+   *
+   * @default - none
+   */
+  readonly timeout?: HttpTimeout;
+
+  constructor(props?: HttpNodeListenerProps) {
+    super();
+    const checkedProps = props ?? {};
+    this.port = checkedProps.port ? checkedProps.port : 8080;
+    this.healthCheck = checkedProps.healthCheck;
+    this.timeout = checkedProps.timeout;
+  }
+
+  /**
+   * Return Listener for HTTP protocol when Listener is added to Virtual Node.
+   */
+  public bind(_scope: cdk.Construct): CfnVirtualNode.ListenerProperty {
+    return {
+      portMapping: {
+        port: this.port,
+        protocol: Protocol.HTTP,
+      },
+      healthCheck: HttpNodeListener.renderHealthCheck({
+        port: this.port,
+        protocol: Protocol.HTTP,
+      }, this.healthCheck),
+      timeout: this.timeout ? HttpNodeListener.renderTimeout(this.timeout) : undefined,
+    };
+  }
+}
+
+/**
+ * Represents the properties required to define a HTTP2 Listener for a VirtualNode.
+ */
+export class Http2NodeListener extends VirtualNodeListener {
+/**
+ * Returns the ListenerTimeoutProperty for HTTP2 protocol
+ */
+  public static renderTimeout(tm: Http2Timeout): CfnVirtualNode.ListenerTimeoutProperty {
+    return ({
+      http2: {
+        idle: tm?.idle !== undefined ? {
+          unit: 'ms',
+          value: tm?.idle.toMilliseconds(),
+        } : undefined,
+        perRequest: tm?.perRequest !== undefined ? {
+          unit: 'ms',
+          value: tm?.perRequest.toMilliseconds(),
+        } : undefined,
+      },
+    });
+  }
+
+  /**
+   * Port to listen for connections on
+   *
+   * @default - 8080
+   */
+  readonly port: number;
+
+  /**
+   * Health checking strategy upstream nodes should use when communicating with the listener
+   *
+   * @default - no healthcheck
+   */
+  readonly healthCheck?: HealthCheck;
+
+  /**
+   * Listener timeout property for HTTP2 protocol
+   *
+   * @default - none
+   */
+  readonly timeout?: Http2Timeout;
+
+  constructor(props?: Http2NodeListenerProps) {
+    super();
+    const checkedProps = props ?? {};
+    this.port = checkedProps.port ? checkedProps.port : 8080;
+    this.healthCheck = checkedProps.healthCheck;
+    this.timeout = checkedProps.timeout;
+  }
+
+  /**
+   * Return Listener for HTTP2 protocol when Listener is added to Virtual Node.
+   */
+  public bind(_scope: cdk.Construct): CfnVirtualNode.ListenerProperty {
+    return {
+      portMapping: {
+        port: this.port,
+        protocol: Protocol.HTTP2,
+      },
+      healthCheck: Http2NodeListener.renderHealthCheck({
+        port: this.port,
+        protocol: Protocol.HTTP2,
+      }, this.healthCheck),
+      timeout: this.timeout ? Http2NodeListener.renderTimeout(this.timeout) : undefined,
+    };
+  }
```

Comment:
From what I can see, this is duplicated from the `HttpNodeListener` class. We need to have a single class that represents those two listeners, that you can pass either `HTTP` or `HTTP2` as the protocol, to get rid of this duplication.